src/pages/Markets/constants.ts

```diff
@@ -7,3 +11,98 @@ export enum MARKETS_CATEGORIES {
   ONE_CLICK_DEFI = 'oneClickDefiAssets',
   THORCHAIN_DEFI = 'thorchainDefi',
 }
+
+export const sortOptionsByCategory: Record<MARKETS_CATEGORIES, SortOption[] | undefined> = {
+  [MARKETS_CATEGORIES.TRADING_VOLUME]: undefined,
+  [MARKETS_CATEGORIES.MARKET_CAP]: [
+    {
+      key: SortOptionsKeys.VOLUME,
+      label: 'dashboard.portfolio.volume',
+    },
+    {
+      key: SortOptionsKeys.PRICE_CHANGE,
+      label: 'dashboard.portfolio.priceChange',
+    },
+    {
+      key: SortOptionsKeys.MARKET_CAP,
+      label: 'dashboard.portfolio.marketCap',
+    },
+  ],
+  [MARKETS_CATEGORIES.TRENDING]: [
+    {
+      key: SortOptionsKeys.VOLUME,
+      label: 'dashboard.portfolio.volume',
+    },
+    {
+      key: SortOptionsKeys.PRICE_CHANGE,
+      label: 'dashboard.portfolio.priceChange',
+    },
+    {
+      key: SortOptionsKeys.MARKET_CAP,
+      label: 'dashboard.portfolio.marketCap',
+    },
+  ],
+  [MARKETS_CATEGORIES.TOP_MOVERS]: [
+    {
+      key: SortOptionsKeys.VOLUME,
+      label: 'dashboard.portfolio.volume',
+    },
+    {
+      key: SortOptionsKeys.PRICE_CHANGE,
+      label: 'dashboard.portfolio.priceChange',
+    },
+    {
+      key: SortOptionsKeys.MARKET_CAP,
+      label: 'dashboard.portfolio.marketCap',
+    },
+  ],
+  [MARKETS_CATEGORIES.RECENTLY_ADDED]: [
+    {
+      key: SortOptionsKeys.VOLUME,
+      label: 'dashboard.portfolio.volume',
+    },
+    {
+      key: SortOptionsKeys.PRICE_CHANGE,
+      label: 'dashboard.portfolio.priceChange',
+    },
+    {
+      key: SortOptionsKeys.MARKET_CAP,
+      label: 'dashboard.portfolio.marketCap',
+    },
+  ],
+  [MARKETS_CATEGORIES.ONE_CLICK_DEFI]: [
+    {
+      key: SortOptionsKeys.APY,
+      label: 'common.apy',
+    },
+    {
+      key: SortOptionsKeys.VOLUME,
+      label: 'dashboard.portfolio.volume',
+    },
+    {
+      key: SortOptionsKeys.MARKET_CAP,
+      label: 'dashboard.portfolio.marketCap',
+    },
+  ],
+  [MARKETS_CATEGORIES.THORCHAIN_DEFI]: [
+    {
+      key: SortOptionsKeys.APY,
+      label: 'common.apy',
+    },
+    {
+      key: SortOptionsKeys.VOLUME,
+      label: 'dashboard.portfolio.volume',
+    },
+    {
+      key: SortOptionsKeys.MARKET_CAP,
+      label: 'dashboard.portfolio.marketCap',
+    },
+  ],
+}
```

Comment:
I kept the filters for each categories so we have a granular control on what we display for each category